### DISCUSSION TITLE
How do I make graphviz use libgio from 'glib' instead of using the system's copy?

### DISCUSSION THREAD

**Discussion by wscott** [Homebrew/discussions#3535] [Linux]
I am trying to install graphviz on SLES 12.

````
ps2pdf $psfile && rm -f $psfile || { rm -f $psfile; exit 1; }
  CC       dot_builtins-dot_builtins.o
  CXXLD    dot_builtins
  CCLD     dot
/nfs/site/disks/aadg.wscott.1/linuxbrew/bin/ld: warning: libselinux.so.1, needed by /usr/lib64/libgio-2.0.so.0, not found (try using -rpath or -rpath-link)
/nfs/site/disks/aadg.wscott.1/linuxbrew/bin/ld: /usr/lib64/libgio-2.0.so.0: undefined reference to `getfilecon_raw'
/nfs/site/disks/aadg.wscott.1/linuxbrew/bin/ld: /usr/lib64/libgio-2.0.so.0: undefined reference to `fgetfilecon_raw'
/nfs/site/disks/aadg.wscott.1/linuxbrew/bin/ld: /nfs/site/disks/aadg.wscott.1/linuxbrew/Cellar/pango/1.50.9/lib/libpango-1.0.so: undefined reference to `fribidi_g
et_par_embedding_levels_ex'
/nfs/site/disks/aadg.wscott.1/linuxbrew/bin/ld: /usr/lib64/libgio-2.0.so.0: undefined reference to `is_selinux_enabled'
/nfs/site/disks/aadg.wscott.1/linuxbrew/bin/ld: /usr/lib64/libgio-2.0.so.0: undefined reference to `lgetfilecon_raw'
````

It looks like I want to be using brew's libgio-2.0 instead of the system copy. So tried making this change:
````
diff --git a/Formula/graphviz.rb b/Formula/graphviz.rb
index 6b5d82da893..812a537cc2c 100644
--- a/Formula/graphviz.rb
+++ b/Formula/graphviz.rb
@@ -33,6 +33,8 @@ class Graphviz < Formula
   on_linux do
     depends_on "byacc" => :build
     depends_on "ghostscript" => :build
+    depends_on "glib" => :build
+    depends_on "python3" => :build
   end
 
   def install
````
(python3 is needed because the system's python is not new enough)

But adding glib here still ends up linking the wrong libgio.  What else do I need to do?

**Comment by danielnachun** [comment]
I am guessing the issue here is opportunistic linkage, where at build time the configuration system picks up that you have `glib` installed and tries to link to it.

If your install prefix is of equal or shorter length than `/home/linuxbrew/.linuxbrew`, you can try to set `HOMEBREW_RELOCATE_BUILD_PREFIX=1` and you should be able to use the bottle even though it's normally not relocatable.  Prefix relocation is an experimental feature right now but from what I've seen `graphviz` does work fine.

If your install prefix is too long or you encounter some breakage, we'll have to take a closer look at what's going on.  But the easiest answer here is to avoid building from source if possible.

**Reply by wscott** [reply]
Unfortunately, I have a long prefix so the relocate hack won't work. This isn't a high priority for me, I was just looking for suggestions for what to try to use the brew's libraries instead of the system ones.
